[하지현] Assignment 6, 7, 8: Express API - CRUD (2)

haaazzi opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 데이터베이스 작업
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 
1. 게시글 수정하기
2. 게시글 삭제하기
3. 좋아요 누르기

<br />

## :: 구현 사항 설명

<br />

## :: 테스트 결과 이미지
1. Terminal
(게시글 수정하기)
![스크린샷 2023-04-20 오후 9 05 54](https://user-images.githubusercontent.com/127677568/233360905-59f6a95d-6290-4ecb-8fd7-bebab10d2241.png)
(게시글 삭제하기)
![스크린샷 2023-04-20 오후 9 07 45](https://user-images.githubusercontent.com/127677568/233361311-cdc86199-60bc-4adc-9c2a-974204e5da38.png)
(좋아요 누르기 / unique 키 작동 O )
![스크린샷 2023-04-20 오후 8 20 02](https://user-images.githubusercontent.com/127677568/233351034-dc65f0a3-d841-445c-a70f-306b10c9becb.png)

2. Postman
(게시글 수정하기)
![스크린샷 2023-04-20 오후 8 43 56](https://user-images.githubusercontent.com/127677568/233356334-078c8306-233c-4ef3-8260-b91990247e6e.png)
(게시글 삭제하기)
![스크린샷 2023-04-20 오후 8 40 47](https://user-images.githubusercontent.com/127677568/233355573-24b3469b-35ff-4c43-aaf8-8cb964d29e81.png)
(좋아요 누르기)
![스크린샷 2023-04-20 오후 8 43 03](https://user-images.githubusercontent.com/127677568/233356000-5e628469-2c77-426d-a0f6-e80419068cf5.png)<br />

Commit:
[하지현] Assignment 6, 7, 8: Express API - CRUD (2)

Jon9Beom commented on jihyeon/app.js:
```diff
@@ -107,6 +78,100 @@ app.post('/users/posts', async function (req, res) {
+    `,[ userId, postId ])
+
+    res.status(201).json({message: 'likeCreated'})
+});
```
post 메소드로 동작하는 하나의 likes api안에서, 기존에 같은 데이터가 likes table내에 존재하는지 확인하여 기존에 있으면 delete query를, 없으면 insert into query를 실행할 수 있도록 로직을 작성해보세요!

Jon9Beom commented on jihyeon/app.js:
```diff
@@ -107,6 +78,100 @@ app.post('/users/posts', async function (req, res) {
+    res.status(201).json({message: 'likeCreated'})
+});
+
+app.get('/posts/viewall', async function (req, res){
+    await dataSource.query(
+        `SELECT
+            u.id AS user_id,
+            p.id AS post_id,
+            p.post_image,
+            p.post_content
+        FROM posts p 
+        INNER JOIN users u ON p.user_id = u.id`
+                , function (err, rows) {
+        res.status(200).json(rows);
+          });
+});
+
```
- 비동기 처리를 하는 방법은 크게 callback, promise, async/await를 사용하는 방법이 있는데, 작성하신 코드에서는 이를 혼합해서 사용하고 있습니다. 아래 예제를 참고해서 리팩토링해주세요.
- 아래 예제에서처럼 callback 방식의 비동기처리 혹은 async/await를 활용한 비동기처리의 두가지 방식 중 하나로 통일되게 리팩토링 하되, 가급적 가장 최근의 문법이자 사용하기 쉽고 가독성이 뛰어난 async/await를 활용하여 수정해주세요.
    
    ```jsx
    // Bad
    app.get("/posts", async (req, res) => {
        await myDataSource.manager.query(
            `SELECT 
                u.id AS userId,
                u.profile_Image AS userProfileImage,
                p.id AS postingId,
                p.title AS postingTitle,
                p.content AS postingContent
            FROM users u
            INNER JOIN posts p ON
            u.id = p.user_id
            `,
            (err, rows) => {
                res.status(200).json({ data: rows });
            }
        );
    });
    
    // Good - callback 방식의 비동기처리
    app.get("/posts", (req, res) => {
        myDataSource.query(`SELECT * FROM users`,
          // 비동기 처리 이후에 실행되는 callback
          (err, data) => {res.status(200).json({ data: data })}
        );
    });
    
    // Good - async와 await 방식의 비동기처리
    app.get("/posts", async(req, res) => {
      const users = await myDataSource.query(`SELECT * FROM users`);
      
      res.status(200).json({ data : users})
    });
    
    ```

Jon9Beom commented on jihyeon/app.js:
```diff
@@ -107,6 +78,100 @@ app.post('/users/posts', async function (req, res) {
+        `, [ userId, postId ]
+    );
+    
+    res.status(200).json({ data : result })
```
```suggestion
    res.status(201).json({ data : result })
```

Jon9Beom commented on jihyeon/app.js:
```diff
@@ -107,6 +78,100 @@ app.post('/users/posts', async function (req, res) {
+    res.status(200).json({ data : result })
+});
+
+app.delete('/users/:userId/posts/delete/:postId', async function (req, res){
```
```suggestion
app.delete('/users/:userId/posts/:postId', async function (req, res){
```
delete라는 동작은 http method를 통해 이미 충분히 설명되므로 지워주세요!

Commit:
MODIFY: app.js
1. likes api 데이터 존재 확인 후 상황에 맞게 delete/insert into query 실행하기
2. async/await 최신 문법으로 수정
3. 알맞는 status code로 수정

Jon9Beom commented:
LGTM 👍
